supplementary_style_guide/style_guidelines/general-formatting.adoc

```diff
@@ -21,3 +21,72 @@ When the format _day Month year_ causes a presentation or clarity issue, use ins
 
 [[product-names-versions-ref]]
 == Product names and version references
+
+////
+PROPOSED ADDITIONS?
+master-docinfo.xml specific and portal-specific version variables
+URL fragments and composed links?
+correct sequence when combining attributes
+ensure proper sequence for compound attributes
+syntax recommendations: what separators to use? dashes/hyphens, underscores?
+////
+
+Whenever you refer to the name of your product in full, or in its abbreviated form, or when you refer to the major and minor version of your product, avoid using hard-coded references and use attributes instead.
+
+Only use hard-coded version references if the version that you are referring to in that particular case never changes.  
+
+=== Attribute file
+
+Define attributes for product name and product version in a single common shared attributes file for each documentation set or product.
+
+Name the file `attributes.adoc`.
+You may use a different name if necessary, but if you do, ensure that the name of the file indicates to your collaborators that it is an attribute file.
+
+//For a single title, store the file in the same directory as the `master.adoc` file.
+For sharing between multiple titles, store the file in a common directory above the directories that contain your titles, such as `docs/common/` or `docs/topics/templates`.
```

Comment:
`docs/artifacts` might be a better example

Reply:
@pwright Thank you. One question: Better than which one of the two that I put there? :)
While both directory path examples I included here are based on examples taken from EAP, Quarkus, CNR and JWS, I am unaware of any project at red hat that stores attribute files in `docs/artifacts`. (You can see that I fell victim to my own bias based on experience.) 
That being said I believe that `docs/common` is as valid an example as any here (Especially considering that `attributes.adoc` is in its nature and intended purpose neither an artifact nor a template).
I will add `docs/artifacts` as another example of a location where `attributes.adoc` can be stored.

Reply:
Another alternative that comes to mind is to avoid giving an example altogether. Writers can be reasonably expected to comprehend the guideline based on what is written down, even without an example.That way we can avoid sounding too prescriptive if any of you are concerned about that. Thoughts?

Reply:
@inoxx03  the reason I suggested this is because that's the standard ;) https://github.com/redhat-documentation/modular-docs/tree/mod-doc-repo-example
But, of course, because I'm not using that standard, I got the example wrong... should have been `docs/_artifacts` 

I think leaving `docs/common` there makes sense, cos it reflects common usage.

Reply:
@inoxx03 I agree with @pwright about leaving it as-is, but just make it clear that this is an example.

Reply:
@pwright Ah, thank you for the clarification:) I did not check the directory recommendation in the mod-docs guide, which is my mistake. I should have done so. Indeed, `docs/_artifacts` seems to me like a directory compatible with Pantheon MS3. Given that we want to recommend a directory structure compatible with the next generation of our publishing tools, using your suggested example seems highly appropriate. I'll add that information.

@stoobie Thank you, I will make it clear in the section that what we list as directory names is to be seen as a recommendation rather than a hard-and-fast rule.

Reply:
IMHO, Stefan uses language (i.e. "such as")  that implies that the directories are examples: "...store the file in a common directory above the directories that contain your titles, such as docs/common/ or docs/topics/templates."